hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/HoodieDeltaStreamer.java

```diff
@@ -326,6 +328,16 @@ private boolean onDeltaSyncShutdown(boolean error) {
     @Parameter(names = {"--checkpoint"}, description = "Resume Delta Streamer from this checkpoint.")
     public String checkpoint = null;
 
+    /**
+     * 1. string: topicName,partition number 0:offset value,partition number 1:offset value
+     * 2. timestamp: kafka offset timestamp
+     * example
+     * 1. hudi_topic,0:100,1:101,2:201
+     * 2. 1621947081
+     */
+    @Parameter(names = {"--checkpoint-type"}, description = "Checkpoint type, divided into timestamp or string offset")
+    public String checkpointType = "string";
```

Comment:
sorry, why do we have this config in two places. We have it defined as top level config in HoodieDeltaStreamer.Config. But in KafkaOffsetGen, I see you are accessing it as "hoodie.deltastreamer.source.kafka.checkpoint.type". May be we should rely on this config param and remove it from top level since this is applicable just to kafka for now.

Reply:
I have not finished processing this PR, so this is currently a semi-finished product. When I finish processing, I will ping you
@nsivabalan